<?xml version="1.0"?>

<launch>

  <arg name="drone_name"            value="ragnarok"/>

  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen">
    <param name="udp" value="false"/>
    <param name="port" value="/dev/ttyACM0"/>
  </node>
  
  <!-- GPS Node-->
  <node name="f9p" pkg="ublox" type="ublox_ros">
    <param name="local_host" value="localhost"/>
    <param name="rover_host" value="localhost"/>
    <rosparam command="load" file="$(find ublox)/params/ublox.yaml"/>
    <param name="serial_port" value="/dev/ttyACM1"/>
    <!--only using a single receiver-->
    <param name="rover_quantity" value="0"/>
    <param name="base_type" value="moving"/>
   </node> 
  
  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/$(arg drone_name).yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
  </node>

  <!-- Estimator -->
  <rosparam command="load" file="$(find roscopter)/params/ekf.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="baro" to="/baro"/>
    <remap from="ublox_gnss" to="/PosVelEcef"/>

  </node>

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
  </node>

</launch>
